ts/Core/Axis/Axis.ts

```diff
@@ -493,6 +493,35 @@ class Axis {
         // List of plotLines/Bands
         axis.plotLinesAndBands = [];
 
+        // Creating labelCollector for plotLines/Bands
+        const collectors = chart.labelCollectors,
+            plotBandCollector = (
+                function (): (SVGElement | undefined)[] {
```

Comment:
Maybe this should be a (static?) class method? That would make things more flexible in terms for extensions of Axis

Reply:
maybe it could be a `defaultCollectors` array/object 🤔

Reply:
It cannot be static because of non-static references to class-instance (`this.options` etc), some strange stuff also happened to `this` references when I simply made it a class method 🤔 , so instead there is now a class factory function, `createCollector` which users then can override and collect whatever labels they want.

Reply:
I was thinking something like:

```ts
    public static defaultCollector(axis: Axis): (SVGElement | undefined)[] {

        // It is possible that plotLinesAndBands and options
        // is/becomes undefined.
        const plotLinesAndBands = (
            !axis?.options?.labels?.allowOverlap &&
                axis?.plotLinesAndBands
        );

        if (plotLinesAndBands) {
            const labels: (SVGElement | undefined)[] = [];

            for (const { label } of plotLinesAndBands) {
                labels.push(label);
            }
            return labels;
        }

        return [];
    }
    /**
     * Overridable function which creates a function collecting labels.
     * The returned function is used for detecting overlapping labels.
     *
     * @function Highcharts.Axis#createCollector
     *
     * @return {Highcharts.LabelCollectorFunction}
     * A function which returns labels.
     */
    public createCollector(): Chart.LabelCollectorFunction {
        return (): (SVGElement | undefined)[] =>
            Axis.defaultCollector(this);
    }

```

Which seems to be working. But just having createCollector works too :)